### PR TITLE
#34906 Set affinity to text for sqlite, if column data type is unknown

### LINKED ISSUE #34906: Some text fields show up as type "NA" and DBeaver can't find text in these fields
### Description

I'm trying to edit my Adobe Lightroom catalog file (SQLite) to change the location where it thinks my photos are. When I use "Search" -> "DB Full TexT" to search for the string "thunder", the search returns no results, despite my being able to see this string just fine in some fields. Further investigation reveals that DBeaver reports the data type for these fields as "NA", which I assume has something to do with the inability to search for text in them. Can anyone help me here?

### DBeaver Version

Enterprise Version 24.1.0.202406100905

### Operating System

macOS 14.5

### Database and driver

SQLite (Pro)

### Steps to reproduce

I've created a tiny version of the original database with just one table in it. Download it from:

https://www.dropbox.com/scl/fi/hwtr77qw78gbkm5jxn7pg/test.lrcat?rlkey=emz9anxkt1oc13vv5oos6600o&st=w4zl7ltm&dl=0

Open a SQLite connection to the file and do a full-text search for "thunder". It returns no results, even though you can see the string in the field "absolutePath"

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by E1izabeth**
Full text search should work now.

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteDataSource.java [hunk 1]
```diff
@@ -87,7 +87,8 @@ public DBSDataType getLocalDataType(String typeName) {
         } else if (typeName.startsWith("REAL") || typeName.startsWith("FLOA") || typeName.startsWith("DOUB")) {
             affinity = SQLiteAffinity.REAL;
         } else {
-            affinity = SQLiteAffinity.NUMERIC;
+            // If type is unknown, let's assume it's a text. Otherwise, search and data editor doesn't work right.
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteDataSource.java:93]: Thus it never can be integer numeric. Add extra if for "INT".

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteDataSource.java [hunk 2]
```diff
@@ -86,8 +86,14 @@ public DBSDataType getLocalDataType(String typeName) {
             affinity = SQLiteAffinity.BLOB;
         } else if (typeName.startsWith("REAL") || typeName.startsWith("FLOA") || typeName.startsWith("DOUB")) {
             affinity = SQLiteAffinity.REAL;
-        } else {
+        } else if (typeName.toUpperCase().contains(SQLConstants.DATA_TYPE_INT) || typeName.toUpperCase().contains("NUMERIC") ||
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteDataSource.java]: Maybe cache the result of `typeName.toUpperCase()`?

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteDataSource.java]: it's already in upper case, actually, so no needed here
fixed

**Review by ShadelessFox — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteDataSource.java [hunk 3]
```diff
@@ -86,8 +86,13 @@ public DBSDataType getLocalDataType(String typeName) {
             affinity = SQLiteAffinity.BLOB;
         } else if (typeName.startsWith("REAL") || typeName.startsWith("FLOA") || typeName.startsWith("DOUB")) {
             affinity = SQLiteAffinity.REAL;
-        } else {
+        } else if (typeName.contains(SQLConstants.DATA_TYPE_INT) || typeName.contains("NUMERIC") || typeName.contains("DECIMAL") ||
+            typeName.contains("DATE") || typeName.contains("TIME") || typeName.contains("BOOL")
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteDataSource.java]: Date-time fields must be strings (they are strings now)

**Review by serge-rider — CHANGES_REQUESTED** [review]

**Comment by uslss** [comment]
verified

**Pull request merged**